website/docs/guides/orchestration/webhooks/zapier-ms-teams.md

```diff
@@ -0,0 +1,153 @@
+ - extract the results from the dbt Cloud admin API 
+ - post a summary to a Microsoft Teams channel
+
+![Screenshot of a message in MS Teams showing a summary of a dbt Cloud run which succeeded](/img/guides/orchestration/webhooks/zapier-ms-teams/ms-teams-ui.png)
+
+## Integration steps:
+### Set up the connection between Zapier and Microsoft Teams 
+
+Install the [Zapier app in Microsoft Teams](https://appsource.microsoft.com/en-us/product/office/WA200002044) and [grant Zapier access to your account](https://zapier.com/blog/how-to-automate-microsoft-teams/). 
+
+Note that when you install the Zapier app, it needs to be added to the team whose channel will receive the message.
+
+### Create a new Zap in Zapier
+Use **Webhooks by Zapier** as the Trigger, and **Catch Raw Hook** as the Event. If you don't intend to [validate the authenticity of your webhook](docs/deploy/webhooks#validate-a-webhook) (not recommended!) then you can choose **Catch Hook** instead. 
+
+Press **Continue**, then copy the webhook URL and put it into dbt Cloud. 
+
+![Screenshot of the Zapier UI, showing the webhook URL ready to be copied](/img/guides/orchestration/webhooks/zapier-common/catch-raw-hook.png)
+
+### Configure a new webhook in dbt Cloud
+See [Create a webhook subscription](/docs/deploy/webhooks#create-a-webhook-subscription) for full instructions. Choose either **Run completed** or **Run errored**, but not both or you'll get double messages when a run fails.
+
+Make note of the Webhook Secret Key for later.
+
+Once you've tested the endpoint in dbt Cloud, go back to Zapier and click **Test Trigger**, which will create a sample webhook body based on the test event dbt Cloud sent.
+
+The sample body's values are hard-coded and not reflective of your project, but they give Zapier a correctly-shaped object during development. 
+
+### Store secrets 
+In the next step, you will need the Webhook Secret Key from the prior step, and a dbt Cloud [user token](https://docs.getdbt.com/docs/dbt-cloud-apis/user-tokens) or [service account token](https://docs.getdbt.com/docs/dbt-cloud-apis/service-tokens). 
+
+Zapier allows you to [store secrets](https://help.zapier.com/hc/en-us/articles/8496293271053-Save-and-retrieve-data-from-Zaps), which prevents your keys from being displayed in plaintext in the Zap code. You will be able to access them via the [StoreClient utility](https://help.zapier.com/hc/en-us/articles/8496293969549-Store-data-from-code-steps-with-StoreClient).
+
+### Add a code action
+Select **Code by Zapier** as the App, and **Run Python** as the Event. 
+
+In the **Set up action** area, add two items to **Input Data**: `raw_body` and `auth_header`. Map those to the `1. Raw Body` and `1. Headers Http Authorization` fields from the **Catch Raw Hook** step above.
+
+![Screenshot of the Zapier UI, showing the mappings of raw_body and auth_header](/img/guides/orchestration/webhooks/zapier-common/run-python.png)
+
+In the **Code** field, paste the following code: 
```

Comment:
I think it would be helpful to have a brief description of what this code is accomplishing, even though you have in-line comments as well.

Reply:
At a very high level, that's what the "this integration will" section up top does - what sort of granularity were you thinking?

Reply:
I think this the "integration will" section is useful for what this guide and implementation are explaining, but I think explaining the code is a little different. Would recommend just abstracting out some of your in-line code comments into a brief sentence or two (because on first glance, code is intimidating!), but again, just a suggestion!